cli/flags.rs

```diff
@@ -1315,35 +1317,80 @@ pub fn resolve_urls(urls: Vec<String>) -> Vec<String> {
 }
 
 /// Expands "bare port" paths (eg. ":8080") into full paths with hosts. It
-/// expands to such paths into 3 paths with following hosts: `0.0.0.0:port`,
-/// `127.0.0.1:port` and `localhost:port`.
+/// expands to such paths into 5 paths with following hosts: `0.0.0.0:port`,
+/// `127.0.0.1:port`, `[::]:port`, `[::1]:port` and `localhost:port`.
 fn resolve_hosts(paths: Vec<String>) -> Vec<String> {
+  lazy_static! {
+    // https://tools.ietf.org/html/rfc1035
+    static ref DOMAIN_ALLOWED_CHARACTERS: Regex = Regex::new(r"^[-.a-zA-Z0-9]+$").unwrap();
+  }
+
   let mut out: Vec<String> = vec![];
   for host_and_port in paths.iter() {
+    // pure port:                 :1234
+    // ipv4 with port:            1.2.3.4:1234
+    // ipv6 with port:            [1:2:3::4]:1234
+    // pure ipv4:                 1.2.3.4
+    // pure ipv6 with bracket:    [1:2:3::4]
+    // pure ipv6 without bracket: 1:2:3::4
+    // pure domain name:          example.com
+    // domain name with port:     example.com:1234
+
+    if host_and_port.parse::<SocketAddr>().is_ok() {
+      // ipv4 with port / ipv6 with port
+      out.push(host_and_port.to_owned());
+      continue;
+    }
+    if host_and_port.parse::<IpAddr>().is_ok() {
+      // pure ipv4 / pure ipv6 without bracket
+      out.push(host_and_port.to_owned());
+      continue;
+    }
+    if host_and_port.starts_with('[')
+      && host_and_port.ends_with(']')
+      && host_and_port[1..host_and_port.len() - 1]
+        .parse::<Ipv6Addr>()
+        .is_ok()
+    {
+      // pure ipv6 with bracket
+      out.push(host_and_port.to_owned());
+      continue;
+    }
+    if host_and_port.starts_with(':') {
+      let port = host_and_port[1..].parse::<i32>().unwrap_or(0);
+      if port > 0 && port < 65536 {
+        // pure port
+        for host in
+          ["0.0.0.0", "127.0.0.1", "[::]", "[::1]", "localhost"].iter()
+        {
+          out.push(format!("{}:{}", host, port));
+        }
+        continue;
+      }
+    }
```

Comment:
Could you move this block of code into its own utility function and add unit tests for it?

Reply:
Done.